### PR TITLE
[bugfix/enhancement]: Support Navidrome structured lyrics

### LINKED ISSUE #311: No playlists in "Add to Playlist" dialog
## Expected Behavior

I can right-click a track, click "Add to Playlist", and select a playlist to add the track to.

## Current Behavior

The "Playlists" box is dead. The cursor doesn't change on hover, and nothing happens when I click in it.

image

In the console, I see:

> renderer.js:2295     GET http://10.10.10.5:7777/api/playlist?_end=0&_order=ASC&_sort=name&_start=0&smart=false 500 (Internal Server Error)

The response payload for that request is:

> {"error":"no such column: smart"}null

## Steps to Reproduce

Right-click a track anywhere in the app and click "Add to Playlist"

## Your Environment

-   Application version (e.g. v0.1.0) : 0.4.0
-   Operating System and version (e.g. Windows 10) : MacOS 1.12.1
-   Server and version (e.g. Navidrome v0.48.0) : Navidrome 0.47.5

### LINKED ISSUE #468: Embedded lyrics show gibberish
## Current Behavior

image

## Your Environment

-   Application version (e.g. v0.1.0) : 0.5.3
-   Operating System and version (e.g. Windows 10) : Ubuntu 23.10
-   Server and version (e.g. Navidrome v0.48.0) : Navidrome 0.51.0

### PR DESCRIPTION
**Pull request by kgarner7**
This PR provides support for OpenSubsonic structured lyrics (Navidrome >= 0.51.0), and also stores server version/features.

Changes:
- `useServerVersion()`: fetches server version on open, and when changed. This is essential for OS clients to know what features are supported
- `useSongLyricsBySong()`: now supports fetching from OS to retrieve all structured lyrics. If multiple lyrics are shown, a select is available

Notes:
- `offset` is currently not used (of note, it has the opposite meaning to what Feishin does)
- could probably have a better interface for language selection?
- Older versions of Navidrome should still be supported

### CODE REVIEW

File: src/renderer/api/jellyfin/jellyfin-controller.ts [hunk 1]
```diff
@@ -946,6 +948,18 @@ const getSongDetail = async (args: SongDetailArgs): Promise<SongDetailResponse>
     return jfNormalize.song(res.body, apiClientProps.server, '');
 };
 
+const getServerInfo = async (args: ServerInfoArgs): Promise<ServerInfo> => {
+    const { apiClientProps } = args;
+
+    const res = await jfApiClient(apiClientProps).getServerInfo();
+
+    if (res.status !== 200) {
+        throw new Error('Failed to get song detail');
```
Comment (jeffvli) [inline src/renderer/api/jellyfin/jellyfin-controller.ts]: This error message should be for server info.

File: src/renderer/api/subsonic/subsonic-controller.ts [hunk 2]
```diff
@@ -368,12 +372,86 @@ const getRandomSongList = async (args: RandomSongListArgs): Promise<RandomSongLi
...
+    const { query, apiClientProps } = args;
+
+    const res = await ssApiClient(apiClientProps).getStructuredLyrics({
+        query: {
+            id: query.songId,
+        },
+    });
+
+    if (res.status !== 200) {
+        throw new Error('Failed to get server extensions');
```
Comment (jeffvli) [inline src/renderer/api/subsonic/subsonic-controller.ts]: This error message should be for the lyrics.

**Review by jeffvli — CHANGES_REQUESTED** [review]

File: src/renderer/features/lyrics/lyrics.tsx [hunk 3]
```diff
@@ -198,18 +202,28 @@ export const Lyrics = () => {
...
                 <ActionsContainer>
+                    {languages.length > 1 && (
+                        <Select
+                            clearable={false}
+                            data={languages}
+                            style={{ bottom: 50, position: 'absolute' }}
+                            value={index.toString()}
+                            onChange={(value) => setIndex(parseInt(value!, 10))}
+                        />
+                    )}
```
Comment (jeffvli) [inline src/renderer/features/lyrics/lyrics.tsx]: This should be moved into the `LyricsActions` component.

**Comment by kgarner7** [comment]
Also resolves #311 with newest change

**Review by jeffvli — APPROVED** [review]

**Pull request merged**